pySNOM/images.py

```diff
@@ -1,118 +1,121 @@
 import numpy as np
 import copy
 from enum import Enum
-import pySNOM.defaults as defaults
+from pySNOM.defaults import Defaults
+from gwyfile.objects import GwyDataField
 
 MeasurementModes = Enum('MeasurementModes', ['None','AFM', 'PsHet', 'WLI', 'PTE', 'TappingAFMIR', 'ContactAFM'])
 DataTypes = Enum('DataTypes', ['Amplitude', 'Phase', 'Topography'])
 ChannelTypes = Enum('ChannelTypes', ['Optical','Mechanical'])
 
 # Full measurement data containing all the measurement channels
-class Data:
-    def __init__(self, filename = None, data = None, parameters = None, mode = "None"):
-        self.filename = filename  # Full path with name
-        # Measurement mode (PTE, PSHet, AFM, NanoFTIR) - Enum MeasurementModes
-        self.mode = MeasurementModes[mode]
-        # Full data from the gwy files with all the channels
-        self.data = data # All channels - Dictionary
-        # Other parameters from info txt -  Dictionary
-        if parameters is not None:
-            self.setParameters(parameters)
-        else:
-            self.parameters = parameters
-
-    # Method to set the actual data read from the gwyddion or gsf files
-    def setData(self, data: dict):
-        self.data = data
-
-    # Method to set measurement mode directly
-    def setMeasurementMode(self, measmode: str):
-        self.mode = MeasurementModes[measmode]
-
-    # Method to retrieve e specific channel from all the measurement channels
-    def getChannelData(self, channelname: str):
+class Measurement:
+    def __init__(self, filename = None, data = None, info = None, mode = "None"):
+        self.filename = filename # Full path with name
+        self.mode = mode # Measurement mode (PTE, PSHet, AFM, NanoFTIR) - Enum MeasurementModes
+        self._data = data # All channels - Dictionary
+        self.info = info
+
+    @property
```

Comment:
Are you sure we need these properties at all? Why not set everything when the class is initialized and then just keep it read-only.

Reply:
The problem arise from the fact how neaspec organizes the measerement data. You have the actual data in gsf or gwy files but it does not contain all the experimental parameters.
There is an additional text file containing other parameters. You dont always use them and sometimes people loose this file. :D So I thought it could be optinally stored when they need it.
But it is optinal.

Reply:
Sorry, I was not clear. Why do you need setters and getters when you could just specify the contents of this class at initialization?

Reply:
For the `info`, I wanted to keep the flexibility to set it later, if the user does not have the info textfile when starting with the data.
For `mode`, my thinking was the same.
For `data`, you are correct, since the setter does not even go anything it is unnecessary. (I remove it).

Question: Do you suggest not having any setter/getter even for `mode` and `info` but the user should reinitialize the object when they want to change?
Noob question: Is there any way to make an attribute truly read-only in python? My understanding so far is that in the best case, you can try to hide them from the user with for example `_attribute` and create setters/getters but they are still accessible by the dot notation.

Reply:
> Question: Do you suggest not having any setter/getter even for `mode` and `info` but the user should reinitialize the object when they want to change?

That is always my starting point. It is always easiest to think about immutable objects. Imagine passing them around: whichever function gets them could modify them, and that becomes spaghetti. 

Then, if object changing is considered common and useful! enough, setters/getters can be added. As a programmer I prefer the pain of reinitializing if it allows me think less. Of course there is a threshold somewhere...

> Is there any way to make an attribute truly read-only in python?

Not really. What you proposed is the best option.